Recipes/Controller/RecipeDetailViewController.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  RecipeDetailViewController.swift
+//  Recipes
+//
+//  Created by krikaz on 3/12/20.
+//  Copyright © 2020 Lambda Inc. All rights reserved.
+//
+
+import UIKit
+
+class RecipeDetailViewController: UIViewController {
+    
+    @IBOutlet weak var recipeLabel: UILabel!
+    @IBOutlet weak var recipeDescription: UITextView!
+    
+    var recipe: Recipe? {
+        didSet {
+            updateViews()
+        }
+    }
+    
+    func updateViews() {
+        if self.isViewLoaded, let recipe = recipe {
+            recipeLabel.text = recipe.name
+            recipeDescription.text = recipe.instructions
+        }
+    }
+    
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        updateViews()
+    }
```

Comment:
Since you're calling `updateViews` from `viewDidLoad`, your view is loaded so there's no need to check for `self.isViewLoaded`.

Another trick is to not use optional binding like `if let recipe = recipe` but use [Optional Chaining](https://docs.swift.org/swift-book/LanguageGuide/OptionalChaining.html) like this :

```swift
recipeLabel.text = recipe?.name
recipeDescription.text = recipe?.instructions
```

It works because the `text` property is an optional String (`String?`). It basically means "If recipe is not nil, use name and instructions otherwise use nil".